Source/JavaScriptCore/runtime/CommonSlowPaths.cpp

```diff
@@ -306,8 +307,16 @@ JSC_DEFINE_COMMON_SLOW_PATH(slow_path_check_tdz)
     auto bytecode = pc->as<OpCheckTdz>();
     if (bytecode.m_targetVirtualRegister == codeBlock->thisRegister())
         THROW(createReferenceError(globalObject, "'super()' must be called in derived constructor before accessing |this| or returning non-object."_s));
-    else
-        THROW(createTDZError(globalObject));
+    else {
+        auto [block, index] = getBytecodeIndex(codeBlock->vm(), callFrame);
+        auto info = block->expressionInfoForBytecodeIndex(index);
+        int expressionStart = info.divot - info.startOffset;
+        int expressionStop = info.divot + info.endOffset;
+
+        RefPtr provider = codeBlock->source().provider();
+        auto ident = provider->getRange(expressionStart, expressionStop).toString();
```

Comment:
I don't think you need to do `.toString()` here. Although you might need to annotate `LIFETIME_BOUND` on `getRange`